### PR TITLE
feat(table): update tree table responsiveness

### LINKED ISSUE #3900: Add responsive/mobile option to Tree Table
Want to add a mobile layout to the tree table component. Preliminary designs here: https://marvelapp.com/prototype/60jcb75

### PR DESCRIPTION
**Pull request by mattnolting**
closes #3900 

Preview: https://patternfly-pr-3943.surge.sh/components/table#tree-table

### CODE REVIEW

**Comment by mattnolting** [comment]
@mcarrano @mceledonia @mcoker the responsive state is built in to the examples, so you'll just need to shrink your viewport width to see the implementation

**Comment by mattnolting** [comment]
@mceledonia @mcarrano updated

**Review by mcarrano — APPROVED** [review]
I like it. Great work @mattnolting . In fact I wonder it we should have used this approach to collapsing row details for all tables on mobile.

**Comment by nicolethoen** [comment]
How's this going, @mattnolting

**Comment by mattnolting** [comment]
> How's this going, @mattnolting

@nicolethoen It's in a finished state, just waiting for reviews.

File: src/patternfly/components/Table/table-tree-view.scss [hunk 1]
```diff
@@ -78,3 +78,176 @@ $pf-c-tree-view--MaxDepth: 10;
...
+
+  .pf-c-table__tree-view-main > .pf-c-table__check {
+    margin-right: var(--pf-c-table--m-tree-view-mobile__tree-view-main--c-table__check--MarginRight);
+    margin-left: var(--pf-c-table--m-tree-view-mobile__tree-view-main--c-table__check--MarginLeft);
+    order: var(--pf-c-table--m-tree-view-mobile__tree-view-main--c-table__check--Order);
+  }
+
+  .pf-c-table__tree-view-details-toggle {
+    .pf-c-button {
+      margin-top: var(--pf-c-table__toggle--c-button--MarginTop);
```
Comment (mcoker) [inline src/patternfly/components/Table/table-tree-view.scss:243]: Typically we would style `.pf-c-table__tree-view-details-toggle` if we know there will always be an action here.

File: src/patternfly/components/Table/table-details-toggle-icon.hbs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="pf-c-table__details-toggle-icon{{#if table-details-toggle-icon--modifier}} {{table-details-toggle-icon--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/Table/table-details-toggle-icon.hbs:1]: needs to be a `<span>`

File: src/patternfly/components/Table/table-tr.hbs [hunk 3]
```diff
@@ -2,6 +2,7 @@
   {{~#if table-tr--expandable}} pf-c-table__expandable-row{{/if}}
   {{~#if table-tr--IsControlRow}} pf-c-table__control-row{{/if}}
   {{~#if table-tr--IsExpanded}} pf-m-expanded{{/if}}
+  {{~#if table-tr--details--IsExpanded}} pf-m-treeview-details-expanded{{/if}}
```
Comment (mcoker) [inline src/patternfly/components/Table/table-tr.hbs:5]: should this be `tree-view`?

```suggestion
  {{~#if table-tr--details--IsExpanded}} pf-m-tree-view-details-expanded{{/if}}
```

File: src/patternfly/components/Table/table-tree-view.scss [hunk 4]
```diff
@@ -78,3 +78,176 @@ $pf-c-tree-view--MaxDepth: 10;
...
+
+  .pf-m-tree-view-grid-2xl.pf-c-table {
+    @media screen and (max-width: $pf-global--breakpoint--2xl) {
+      @content;
+    }
+  }
+}
+
+@include pf-tree-view-mobile-layout {
+  --pf-c-table--pf-m-tree-view-mobile--tr--OutlineOffset: calc(-1 * var(--pf-global--spacer--xs));
```
Comment (mcoker) [inline src/patternfly/components/Table/table-tree-view.scss:119]: should any `pf-c-table--pf-m-tree-view-mobile` var be `pf-c-table--pf-m-tree-view-grid`?

File: src/patternfly/components/Table/table-tree-view.scss [hunk 5]
```diff
@@ -78,3 +78,176 @@ $pf-c-tree-view--MaxDepth: 10;
...
+  --pf-c-table--pf-m-tree-view-mobile--tr--before--BorderColor: var(--pf-global--BorderColor--100);
```
Comment (mcoker) [inline src/patternfly/components/Table/table-tree-view.scss:120]: doesn't look like this is used

File: src/patternfly/components/Table/table-tree-view.scss [hunk 6]
```diff
@@ -78,3 +78,176 @@ $pf-c-tree-view--MaxDepth: 10;
...
+  --pf-c-table--pf-m-tree-view-mobile--tr--before--BorderBottomWidth: var(--pf-global--BorderWidth--sm);
```
Comment (mcoker) [inline src/patternfly/components/Table/table-tree-view.scss:121]: not used

File: src/patternfly/components/Table/examples/Table.md [hunk 7]
```diff
@@ -1536,9 +1536,13 @@ Note: To apply padding to `.pf-c-table__expandable-row`, wrap the content in `.p
 
 | Class | Applied | Outcome |
 | -- | -- | -- |
-| `.pf-c-table__tree-view-main` | `<div>` | Initiates a tree view table main container. |
-| `.pf-c-table__tree-view-content` | `<div>` | Initiates a tree view table content element. |
-| `.pf-c-table__tree-view-icon` | `<span>` | Initiates a tree view icon wrapper. |
+| `.pf-c-table__tree-view-main` | `<div>` | Initiates a tree view table main container. **Required with tree view** |
+| `.pf-c-table__tree-view-content` | `<div>` | Initiates a tree view table content element. **Required with tree view** |
```
Comment (mcoker) [inline src/patternfly/components/Table/examples/Table.md:1540]: I might swap `-content` and `-main`? I'm not sure if we have both `-content` and `-main` in another component, but typically `-main` wraps the title and whatever may go beside it in an item (eg, menu item, tree view/dual list selector node). WDYT?

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks great! Just a handful of comments.

File: src/patternfly/components/Table/table-tree-view-details-toggle.hbs [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<span class="pf-c-table__tree-view-details-toggle{{#if table-tree-view-details-toggle--modifier}} {{table-tree-view-details-toggle--modifier}}{{/if}}"
+  {{#if table-tree-view-details-toggle--attribute}}
+    {{{table-tree-view-details-toggle--attribute}}}
+  {{/if}}>
+  <button class="pf-c-button pf-m-plain{{#if table-tr--details--IsExpanded}} pf-m-expanded{{/if}}" {{#if table-td--button--attribute}}{{{table-td--button--attribute}}}{{/if}} {{#if table-tr--details--IsExpanded}}aria-expanded="true"{{/if}}>
```
Comment (mcoker) [inline src/patternfly/components/Table/table-tree-view-details-toggle.hbs:5]:
```suggestion
  <button class="pf-c-button pf-m-plain{{#if table-tr--details--IsExpanded}} pf-m-expanded{{/if}}" {{#if table-td--button--attribute}}{{{table-td--button--attribute}}}{{/if}} {{#if table-tr--details--IsExpanded}}aria-expanded="true"{{/if}} type="button">
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Do you need these styles for `__tree-view-content`? 

https://github.com/patternfly/patternfly/blob/5500d9d243f4df92e64ca99aa8a941b40d6699d1/src/patternfly/components/Table/table-tree-view.scss#L73-L75

**Comment by mattnolting** [comment]
@mcoker updated

**Review by mcoker — APPROVED** [review]
L🍕TM!!! Great work!! 🥳

**Pull request merged**